File: InvotraxApp/backend/InvotraxApp/src/main/java/dev/lkeleti/invotraxapp/dto/ProductDto.java
package dev.lkeleti.invotraxapp.dto;

import dev.lkeleti.invotraxapp.model.ProductType;
import dev.lkeleti.invotraxapp.model.VatRate;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class ProductDto {
    private Long id;
    private String name;
    private String sku;
    private String description;
    private ProductCategoryDto category;
    private ManufacturerDto manufacturer;
    private BigDecimal netPrice;
    private BigDecimal grossPrice;
    private Integer warrantyPeriodMonths;
    private boolean serialNumberRequired;
    private int stockQuantity;
    private ProductType productType;
    private List<BarcodeDto> barcodes;
    private List<SerialNumberDto> serialNumbers;
    private VatRate vatRate;
    private boolean deleted;
}

File: InvotraxApp/frontend/invotrax/src/app/models/product.dto.ts
import { Decimal } from 'decimal.js';
import { ProductTypeDto } from './productType.dto';
import { ProductCategoryDto } from './product-category.dto';
import { ManufacturerDto } from './manufacturer.dto';
import { SerialNumberDto } from './serial-number.dto';
import { BarcodeDto } from './barcode.dto';
import { VatRateDto } from './vat-rate.dto';

export interface ProductDto {
    id: number;
    name: string;
    sku: string;
    description: string;
    productType: ProductTypeDto;
    category: ProductCategoryDto;
    manufacturer: ManufacturerDto;
    netPurchasePrice: Decimal;
    grossPurchasePrice: Decimal;
    netSellingPrice: Decimal;
    grossSellingPrice: Decimal;
    warrantyPeriodMonths: number;
    serialNumberRequired: boolean;
    stockQuantity: number;
    barcodes: BarcodeDto[];
    serialNumbers: SerialNumberDto[]
    vatRate: VatRateDto;
    deleted: boolean;
}